### ISSUE TITLE
Xcode 12 GM warnings when updating local state from AppState

### ISSUE THREAD

**Issue by NeverwinterMoon** [nalexn/clean-architecture-swiftui#38]
Not sure how critical this is, I just jumped into Xcode 12 GM today, but I have loads of warnings which were not there with Xcode 11. Once again, not sure it actually affects anything, but maybe worth mentioning.

Basically, all the local state updates which come from the injected AppState updates are marked like this:

![image](https://user-images.githubusercontent.com/1917308/93330309-47309280-f827-11ea-915b-a93f3ef5b4ea.png)

**Comment by nalexn** [comment]
These warnings cannot be ignored, we should find the cause.

I can assume you have some code in your project that updates the AppState while the view is being rendered: an [example](https://www.hackingwithswift.com/quick-start/swiftui/how-to-fix-modifying-state-during-view-update-this-will-cause-undefined-behavior).

I double-checked the sample project with the new Xcode and it doesn't throw these warnings, and using `onReceive` for updating the local view state should be perfectly legit.

You can try to put a breakpoint in the AppState update to find the call trace originating from SwiftUI render cycle

**Comment by NeverwinterMoon** [comment]
My only culprit for something like you mentioned would be the use of willSet/didSet, but I don't even have those around the view I get the warnings at. But I did manage to find a connection between those warnings and the navigation. When I navigate to the view and do anything around it, there are no warnings. When I navigate one step deeper down the navigation stack and then return to the view in question, I immediately get those warnings.

**Comment by NeverwinterMoon** [comment]
OK, I did tangle everything together. So I'll go step by step.

First of all, I have this navigation: `View1` -> `View2` -> `View3` -> `View4`

## The original problem
The warnings - they are actually caused by having this on `View2`:

```
.onAppear {
   $someStateVar.wrappedValue = someValueFromUserDefaults
}
```

I also tried to wrap the above in `DispatchQueue.main.async` and it didn't help.

Interestingly enough, when I go: `View1` -> `View2`, there are no warnings.
When I got `View1` -> `View2` -> `View3` -> back to `View2`, the warnings pop up.
Most importantly, this is only with iOS 14, no warnings on iOS 13 with exactly the same code.

## The navigation problem

This one is a bummer. I managed to reproduce it in your project as well, as I wrote above. It's specific only to iOS 14 and works just fine with iOS 13. Basically, if the navigation is deeper than 3 views, the routing approach (with the routingBinding) breaks.

I am fine with using the new to iOS 14 `onOpenURL`, as it allows to attach URL handling logic granularly to every view, which is great, but I still need to support iOS 13 somehow... So I am a bit stuck.

**Comment by nalexn** [comment]
Yeah, I was hoping programmatic navigation will become less buggy with the iOS update, but it looks like things got only worse.

I'll experiment more with your commit and try to find a workaround. I can see one way to unblock you if you're actively working on some project, but this involves quite some refactoring.

I recently wrote an [article](https://nalexn.github.io/uikit-switfui/), where I concluded that for now, I wouldn't trust SwiftUI with programmatic routing and suggested to use UIKit for navigation, while keeping all the screens written in pure SwiftUI. Consider taking this approach. There is also a [project](https://github.com/nalexn/uikit-swiftui) for reference.

**Comment by NeverwinterMoon** [comment]
Looks like a nice approach, that https://github.com/nalexn/uikit-swiftui. I wish I'd used that to begin with. I'll try it out at some point and see how it fits the project.

For now, I went with something like `@ObservedObject private var routing = GlobalRouting()` just for iOS 14. That was an easy change to make...
